pysd/vensim2py.py

```diff
@@ -390,13 +390,13 @@ def parse_general_expression(element, namespace=None, subscript_dict=None, macro
             element['py_name'], element['subs'], expr, init, subscript_dict),
 
         "delay1": lambda in_var, dtime: builder.add_n_delay(
-            in_var, dtime, '0', '1', element['subs'], subscript_dict),
+            in_var, dtime, in_var, '1', element['subs'], subscript_dict),
```

Comment:
Interesting, I think that this way is not working, but I check Delay & Smoth implementation and it really should be work fine. I fix initialization for xmile too. Thanks!

Reply:
Why are you saying that its not working? 
I did notice that the delay test was failing due to a delay with variable order in the model "test_delays.mdl", as far as I understand PySD doesn't support changing order on the go because it would need to dynamicaly create new states. So I still expect this test to fail.

Did you see any behavior different than what I described?

Reply:
No, I say, that this works as required by specification :+1: